api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -869,16 +869,57 @@ type AgentNodePoolPlatform struct {
 	AgentLabelSelector *metav1.LabelSelector `json:"agentLabelSelector,omitempty"`
 }
 
-type AzureNodePoolPlatform struct {
-	VMSize string `json:"vmsize"`
-	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used:
-	// subscription/$subscriptionID/resourceGroups/$resourceGroupName/providers/Microsoft.Compute/images/rhcos.x86_64.vhd
+type AzureVMImageType string
+
+const (
+	ImageID          AzureVMImageType = "ImageID"
+	AzureMarketplace AzureVMImageType = "AzureMarketplace"
+)
+
+// AzureVMImage represents the different types of image data that can be provided for an Azure VM.
+// +union
+type AzureVMImage struct {
+	// Type is the type of image data that will be provided to the Azure VM. This can be either "ImageID" or
+	// "MarketplaceImageInfo".
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=ImageID;AzureMarketplace
+	// +unionDiscriminator
+	Type AzureVMImageType `json:"azureImageType"`
+
+	// ImageID is the Azure resource ID of a VHD image to use to boot the Azure VMs from.
+	//
 	// +optional
+	// +unionMember, optional
 	ImageID string `json:"imageID,omitempty"`
-	// +kubebuilder:default:=120
+
+	// MarketplaceImageInfo contains the Azure Marketplace image info to use to boot the Azure VMs from.
+	//
+	// +optional
+	// +unionMember, optional
+	MarketplaceImageInfo *MarketplaceImage `json:"marketplaceImage,omitempty"`
+}
+
+type AzureNodePoolPlatform struct {
+	// VMSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	//
+	// +kubebuilder:validation:Required
+	VMSize string `json:"vmsize"`
+
+	// Image is the image to boot the VMs with
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Image is immutable"
+	// +kubebuilder:validation:Required
+	Image AzureVMImage `json:"image"`
+
+	// DiskSizeGB is the size in GB to assign to the OS disk
+	// CAPZ default is 30GB, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b3708019a67ff19407b87d63c402af94ca4246f6/api/v1beta1/types.go#L599
+	//
+	// +kubebuilder:default:=30
```

Comment:
this default value change seems is sneaking in, what's the rationale?

Reply:
I copied and pasted that in when I was updating the API fields so that the unit tests would pass. I set this back to what the default was before.